<?xml version="1.0"?>

<launch>
    <arg name="joy_dev" default="/dev/input/js0" />

    <node name="joy" pkg="joy" type="joy_node" output="screen" >
        <param name="dev" value="$(arg joy_dev)" />
        <param name="autorepeat_rate" value="100" />
        <param name="coalesce_interval" value="0.01" />
    </node>

    <arg name="control_mode" default="velocity_altitude"/>

    <arg name="thrust_max" default="0.25"/>
    <arg name="pitch_max" default="0.25"/>
    <arg name="roll_max" default="0.25"/>
    <arg name="yawrate_max" default="0.25"/>

    <arg name="x_dot_max" default="0.25"/>
    <arg name="y_dot_max" default="0.25"/>

    <arg name="initial_x_cmd" default="0.0"/>
    <arg name="initial_y_cmd" default="0.0"/>
    <arg name="initial_z_cmd" default="-1.0"/>
    <arg name="min_z_cmd" default="-10.0"/>
    <arg name="max_z_cmd" default="10.0"/>
    <arg name="z_scale" default="0.1"/>

    <arg name="invert_x" default="false"/>

    <node name="reef_teleop" pkg="reef_teleop" type="reef_teleop" output="screen">
        <rosparam subst_value="true">
            control_mode: $(arg control_mode)

            invert_x: $(arg invert_x)
            x_axis: 3
            y_axis: 4
            z_axis: 2
            thrust_axis: 2
            yaw_axis: 1

            initial_x_pose: $(arg initial_x_cmd)
            initial_y_pose: $(arg initial_y_cmd)
            initial_z_cmd: $(arg initial_z_cmd)
            min_z_cmd: $(arg min_z_cmd)
            max_z_cmd: $(arg max_z_cmd)
            z_scale: $(arg z_scale)

            pitch_max: $(arg pitch_max)
            roll_max: $(arg roll_max)
            thrust_max: $(arg thrust_max)
            yawrate_max: $(arg yawrate_max)

            x_dot_max: $(arg x_dot_max)
            y_dot_max: $(arg y_dot_max)
        </rosparam>
    </node>
</launch>
